client/src/app/client.service.ts

```diff
@@ -0,0 +1,131 @@
+import { Injectable } from '@angular/core';
+import { Observable, of, BehaviorSubject } from 'rxjs';
+import { MessageService } from './message.service';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { catchError, map, tap } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ClientService {
+  private defaultHeaders: HttpHeaders;
+  private subject = new BehaviorSubject<any>(undefined);
+
+  //private usersUrl = 'api/users';  // URL to web api
+  private baseUrl = 'localhost:8999';
+  private serverUrl = 'http://' + this.baseUrl; // URL to server
+
+  private webSocket = new WebSocket('ws://' + this.baseUrl);
+
+  constructor(
+    private http: HttpClient,
+    private messageService: MessageService
+  ) {
+    this.defaultHeaders = new HttpHeaders().set(
+      `Content-Type`,
+      `application/json`
+    );
+
+    this.webSocket.onmessage = (event) => this.handleMessage(event);
+
+    this.subject.subscribe((data) => {
+      console.log(data);
+    });
+  }
+
+  public getSubject(): BehaviorSubject<any> {
```

Comment:
You should avoid returning a `BehaviorSubject` unless it is used in the same class... Return an `Observable` instead